Source/objects.cpp

```diff
@@ -2371,7 +2371,7 @@ void ObjChangeMapResync(int x1, int y1, int x2, int y2)
 
 	for (j = y1; j <= y2; j++) {
 		for (i = x1; i <= x2; i++) {
-			ObjSetMini(i, j, (unsigned char)pdungeon[i][j]);
+			ObjSetMini(i, j, (BYTE)pdungeon[i][j]);

```

Comment:
From PSX sym:
```cpp
extern unsigned char pdungeon[40][40];
```

So, I think we can update the definition to `extern BYTE pdungeon[DMAXX][DMAXY]` and remove this type cast?